rerpc_test.go

```diff
@@ -760,3 +760,101 @@ func TestCallMetadataIntegration(t *testing.T) {
 	assert.Nil(t, err, "call error")
 	assert.Equal(t, res, &pingpb.PingResponse{}, "call response")
 }
+
+//lint:ignore U1000 skip tests while the implementation is WIP
+type headerPingServer struct {
+	pingpb.UnimplementedPingServiceReRPC
+}
+
+func (s headerPingServer) Ping(ctx context.Context, req *pingpb.PingRequest) (*pingpb.PingResponse, error) {
+	handlerMD, _ := rerpc.HandlerMeta(ctx)
+	u := strings.ToUpper(handlerMD.Request().Get("Test-Ping"))
+	handlerMD.Response().Set("Test-Ping", u)
+	return &pingpb.PingResponse{Number: req.Number, Msg: req.Msg}, nil
+}
+
+//lint:ignore U1000 skip tests while the implementation is WIP
+type headerPingInterceptor struct{}
+
+func (i headerPingInterceptor) Wrap(next rerpc.Func) rerpc.Func {
+	return rerpc.Func(func(ctx context.Context, msg proto.Message) (proto.Message, error) {
+		callerMD, _ := rerpc.CallMeta(ctx)
+		requestMeta := callerMD.Request()
+		v := requestMeta.Get("Test-Ping") + "bar"
+		requestMeta.Set("Test-Ping", v)
+		return next(ctx, msg)
+	})
+}
+
+func (i headerPingInterceptor) WrapStream(next rerpc.StreamFunc) rerpc.StreamFunc {
+	return next
+}
```

Comment:
Unless we anticipate more complexity here, this interceptor might be more concise if we use `UnaryInterceptorFunc`.